### PR TITLE
'hf mf sim', 'hf 14a sim' online and offline ar/nr nonces collection

### PR DESCRIPTION
**Pull request by McEloff**
1) Added interactive mode for 'hf 14a sim'
2) Optimized online (interactive mode) and offline process of ar/nr nonces collections for 'hf 14a sim' and 'hf mf sim'
3) Fix AR/NR nonces collection algorithm  in previously merged Mifare1ksim() method
4) Performs postprocess key attack after tag simulation finished
5) Extends offline array up to 32 ar/nr collection per tag simulation

### CODE REVIEW

File: armsrc/iso14443a.c [hunk 1]
```diff
@@ -1390,7 +1389,31 @@ void SimulateIso14443aTag(int tagType, int flags, uint8_t *data) {
...
+                     ar_nr_collected[i].ar,    //AR1
+                     ar_nr_collected[i].nonce2,//NT2
+                     ar_nr_collected[i].nr2,   //NR2
+                     ar_nr_collected[i].ar2    //AR2
+                    );
+        }
+    }
+    // interrupt interactive client
+    if ((flags & FLAG_INTERACTIVE) == FLAG_INTERACTIVE)
+        cmd_send(CMD_ACK, 1, 0, 0, 0, 0);
```
Comment (iceman1001) [inline armsrc/iso14443a.c]: I removed this printing years ago,   I see no reason to bring it back. 
hence the need for flag_interactive is also useless.

**Comment by iceman1001** [comment]
A interesting approach would to drop all those prints,  
and save those nonces in the flashmem,   and download at a given command to run the keyrecovery.

**Comment by McEloff** [comment]
You are right! Flashmem is better for this purpose. My PM V3 without extended flash memory, then I can not release given features now. Let close pull request.

**Comment by iceman1001** [comment]
@McEloff  I thought you had a RDV4.   My bad.

**Comment by pwpiwi** [comment]
No need to use flashmem. There is RAM on all revisions of PM3.

**Comment by McEloff** [comment]
Only for RDV4 platform developing compiler has definition 'WITH_FLASH'. This condition provides access to library for reading and writing individual data variables, structs and arrays from and to various locations. This condition cannot be applied to developing RDV3 and below platforms (V3 devices crashes on startup). There was an idea to use flashmem as non-volatile storage when device working offline instead of printing results back to client after reconnecting.

**Comment by iceman1001** [comment]
@McEloff  correct,     use the hardware advantages of rdv4.

**Comment by pwpiwi** [comment]
Or support all platforms. Flash-Memory is on all devices.

**Comment by iceman1001** [comment]
Wrong repo for that.  This is the dedicated rdv4 repo.  There is too many different  flashmem sizes for the MCU to be useful.  Feel free to add that offical repo.

**Pull request closed without merging**